src/main/resources/templates/java/test/testFiles/behavior/BehaviorTest.java

```diff
@@ -81,15 +91,53 @@ protected Object newInstance(String qualifiedClassName, Object... constructorArg
         return null;
     }
 
+    /**
+     * Instantiate an object of a given class by its qualified name and the constructor arguments, if applicable.
+     * @param constructorArgs: Parameter instances of the constructor of the class, that it has to get instantiated with. Do not include, if the constructor has no arguments.
+     * @return The instance of this class.
+     */
+    protected Object newInstance(Constructor<?> constructor, Object... constructorArgs) {
+        String failMessage = "Could not instantiate the "
+                + constructor.getDeclaringClass() + " because";
+
+        try {
+            return constructor.newInstance(constructorArgs);
+        } catch (IllegalAccessException iae) {
+            fail(failMessage + " access to its constructor with the parameters: "
+                    + getParameterTypesAsString(constructor.getParameterTypes())
+                    + " was denied."
+                    + " Make sure to check the modifiers of the constructor.");
+        } catch (IllegalArgumentException iae) {
+            fail(failMessage + " the actual constructor or none of the actual constructors of this class match the expected one."
+                    + " We expect, amongst others, one with "
+                    + getParameterTypesAsString(constructor.getParameterTypes())
+                    + " parameters, which is not exist."
+                    + " Make sure to implement this constructor correctly.");
+        } catch (InstantiationException ie) {
+            fail(failMessage + " the class is abstract and should not have a constructor."
+                    + " Make sure to remove the constructor of the class.");
+        } catch (InvocationTargetException ite) {
+            fail(failMessage + " the constructor with " + constructorArgs.length + " parameters threw an exception and could not be initialized."
+                    + " Make sure to check the constructor implementation.");
+        } catch (ExceptionInInitializerError eiie) {
+            fail(failMessage + " the constructor with " + constructorArgs.length + " parameters could not be initialized.");
+        } catch (SecurityException se) {
+            fail(failMessage + " access to the package of the class was denied.");
+        }
+
+        return null;
+    }
+
     /**
      * Retrieve an attribute value of a given instance of a class by the attribute name.
      * @param object: The instance of the class that contains the attribute.
      * @param attributeName: The name of the attribute whose value needs to get retrieved.
      * @return The instance of the attribute with the wanted value.
      */
     protected Object valueForAttribute(Object object, String attributeName) {
-        String failMessage = "Could not retrieve the attribute '" + attributeName + "' from the class '"
-            + object.getClass().getSimpleName() + "' because";
+        requireNonNull(object, "receiver must not be null");
```

Comment:
I don't think that `requireNonNull`. How would the feedback in Artemis look like that students get to see in such a case? I think we should rather provide a good fail message.